frontend/http/parser.go

```diff
@@ -128,7 +131,31 @@ func ParseScrape(r *http.Request, opts ParseOptions) (*bittorrent.ScrapeRequest,
 
 	infoHashes := qp.InfoHashes()
 	if len(infoHashes) < 1 {
-		return nil, bittorrent.ClientError("no info_hash parameter supplied")
+		// This is a potential fullscrape. Determine the origin IP and check if
+		// it's whitelisted.
+		host, _, err := net.SplitHostPort(r.RemoteAddr)
+		if err != nil {
+			return nil, err
+		}
+		remoteIP := net.ParseIP(host)
+		if remoteIP == nil {
+			return nil, errors.New("unable to parse HTTP remote address as IP")
+		}
+
+		found := false
+		for _, whitelistedIP := range opts.FullScrapeWhitelistedIPs {
```

Comment:
This code would change if we use CIDR masks, but ideally this is a set lookup and not iteration.